scripts/apps/authoring/authoring/directives/ItemCarouselDirective.ts

```diff
@@ -152,7 +154,7 @@ export function ItemCarouselDirective(notify) {
                 carousel.trigger('to.owl.carousel', [index]);
             };
 
-            function canAddImage(image: IArticle): boolean {
+            function canAddImage(image: IArticle, files?: FileList): boolean {
```

Comment:
Try refactoring this so it's more similar to the diagram I referenced in another comment. Make this function `canAddImages(images: Array<IArticle>): boolean`. Make it possible to accept multiple images and remove reference to files.